### PR TITLE
Revise reduction operations base classes

### PR DESCRIPTION
**Pull request by ggalieroc**
Details: Refactored ArithmeticReductionKeepDims and LogicalReductionKeepDims to use common code for shape inference

- Introduced ReductionBase class to remove duplicated code in output shape inference
- Added checks to axes input shape
- Added check for et of data and axes inputs
- Added type_prop test for all reduction operations
- MVN fusion tests: Element type of axes input of ReduceMean op changed to integer type
- LPT Normalize2 tests: Element type of axes input of Reduce ops changed to integer type

Tickets: 37471

### CODE REVIEW

**Review by blesniewski — APPROVED** [review]
Nitpick: many files lack a newline at file end.

File: ngraph/core/src/op/util/arithmetic_reductions_keep_dims.cpp [hunk 1]
```diff
@@ -32,61 +32,30 @@ bool ngraph::op::util::ArithmeticReductionKeepDims::visit_attributes(AttributeVi
...
+                          axes_et);
+
+    const Rank axes_rank = axes_shape.rank();
+    NODE_VALIDATION_CHECK(this,
+                          axes_rank.compatible(0) || axes_rank.compatible(1),
+                          "Axes input must be a scalar or 1D input. Got: ",
+                          axes_shape);
+
+    PartialShape result_shape =
+        infer_reduction_output_shape(this, data_shape, axes_shape, m_keep_dims);
```
Comment (ilyachur) [inline ngraph/core/src/op/util/arithmetic_reductions_keep_dims.cpp:58]: If we need to pass `this` as a first parameter, can we introduce `infer_reduction_output_shape` as a protected method in the base class?

File: inference-engine/tests/functional/inference_engine/transformations/mvn_fusion_test.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ TEST(TransformationTests, MVNFusionTestOutside) {
         auto sub2 = std::make_shared<ngraph::opset6::Subtract>(input, mean2);
         auto const_2 = ngraph::opset6::Constant::create(ngraph::element::f32, ngraph::Shape{}, { 2 });
         auto power_sqr = std::make_shared<ngraph::opset6::Power>(sub2, const_2);
-        auto mean3_axes = ngraph::opset6::Constant::create(ngraph::element::f32, ngraph::Shape{ 3 }, { 1, 2, 3 });
+        auto mean3_axes = ngraph::opset6::Constant::create(ngraph::element::i32, ngraph::Shape{ 3 }, { 1, 2, 3 });
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/transformations/mvn_fusion_test.cpp:33]: Why didi you change precisions?

File: ngraph/core/src/op/util/arithmetic_reductions_keep_dims.cpp (hunk 1 above)
Comment (ggalieroc) [reply ngraph/core/src/op/util/arithmetic_reductions_keep_dims.cpp]: Done. Please take a look.

File: inference-engine/tests/functional/inference_engine/transformations/mvn_fusion_test.cpp (hunk 2 above)
Comment (ggalieroc) [reply inference-engine/tests/functional/inference_engine/transformations/mvn_fusion_test.cpp:33]: As we discussed in email thread, in these tests nGraph is built manually and for ReduceMean op, axes precision is incorrectly set as f32, while it does not have much sense as axes comprises indexes of input dimension. It is basically the same reason why LPT test precisions were changed.

File: ngraph/core/src/op/util/arithmetic_reduction.cpp [hunk 3]
```diff
@@ -46,24 +46,27 @@ void op::util::ArithmeticReduction::set_reduction_axes(const AxisSet& reduction_
...
 
-    PartialShape result_shape{PartialShape::dynamic()};
+    if (data_rank.is_static() && keep_dims)
+    {
+        result_ps = PartialShape::dynamic(data_rank);
+    }
 
-    auto axes = get_constant_from_source(input_value(1));
-    if (input_rank.is_static() && axes)
+    const auto& axes = get_constant_from_source(this->input_value(1));
```
Comment (ilyachur) [inline ngraph/core/src/op/util/arithmetic_reduction.cpp:60]: nitpick: `this->` is redundant 😄 

Maybe we can use one common `if` for static rank:
```
if (data_rank.is_static()) {
if (keep_dims) {}
if (axes) {}
}
```

File: ngraph/core/src/op/util/logical_reduction.cpp [hunk 4]
```diff
@@ -51,34 +51,27 @@ void op::util::LogicalReduction::set_reduction_axes(const AxisSet& reduction_axe
             ->output(0));
 }
 
-void op::util::LogicalReduction::validate_and_infer_types()
+PartialShape op::util::LogicalReduction::infer_reduction_output_shape(const bool keep_dims)
```
Comment (ilyachur) [inline ngraph/core/src/op/util/logical_reduction.cpp:54]: Looks like this method do the same thing that `ArithmeticReduction::infer_reduction_output_shape` am I right? Can we reuse it?

Comment (ggalieroc) [reply ngraph/core/src/op/util/logical_reduction.cpp]: After we discussed I have introduced a new ReductionBase class with protected method to infer output shape. Please check this [commit](https://github.com/openvinotoolkit/openvino/pull/5609/commits/22f382e42f4e0692d0ed39194649eb3f2af8b1cb)

**Review by ilyachur — APPROVED** [review]

**Pull request merged**